### PR TITLE
Clarify documentation of ssl_set_own_cert()

### LINKED ISSUE #507: Server handshakes with mismatched public/private key pair
mbedTLS does not verify that the public/private key pair matches when doing a handshake, resulting in client side public key signature verification errors if they don't match. This was tested on 2.2.1.

If this is the desired behavior then the documentation of `mbedtls_ssl_conf_own_cert()` should be updated to mention this fact and also point to `mbedtls_pk_check_pair()`.

### PR DESCRIPTION
**Pull request by mpg**
fixes #507 - see issue description

### CODE REVIEW

File: include/mbedtls/ssl.h [hunk 1]
```diff
@@ -2043,6 +2043,14 @@ void mbedtls_ssl_conf_ca_chain( mbedtls_ssl_config *conf,
  *                 whether it matches those preferences - the server can then
  *                 decide what it wants to do with it.
  *
+ * \note           The provided \p pk_key needs to match the public key in the
+ *                 first certificate in \p own_cert, or all handshakes using
+ *                 that certificate will fail. It is your reponsibility
+ *                 to ensure that; this function will not perform any check.
```
Comment (RonEld) [inline include/mbedtls/ssl.h]: I believe the semicolon(`;`) is a typo

Comment (hanno-arm) [reply include/mbedtls/ssl.h]: I think the semicolon is justified here, as the line of thought continues.

Comment (RonEld) [reply include/mbedtls/ssl.h]: yes, the line of thought continues. I think there is no need for the semicolon at all

Comment (hanno-arm) [reply include/mbedtls/ssl.h]: There's not much ground for an argument here, it's a matter of taste. Let's have @mpg decide it.

File: include/mbedtls/ssl.h [hunk 2]
```diff
@@ -2043,6 +2043,14 @@ void mbedtls_ssl_conf_ca_chain( mbedtls_ssl_config *conf,
  *                 whether it matches those preferences - the server can then
  *                 decide what it wants to do with it.
  *
+ * \note           The provided \p pk_key needs to match the public key in the
+ *                 first certificate in \p own_cert, or all handshakes using
+ *                 that certificate will fail. It is your reponsibility
```
Comment (hanno-arm) [inline include/mbedtls/ssl.h]: Typo: `reponsibility`

File: include/mbedtls/ssl.h (hunk 1 above)
Comment (RonEld) [reply include/mbedtls/ssl.h]: I agree

Comment (mpg) [reply include/mbedtls/ssl.h]: FWIW, the semicolon was intentional.

File: include/mbedtls/ssl.h (hunk 2 above)
Comment (mpg) [reply include/mbedtls/ssl.h]: Fixing.

**Comment by mpg** [comment]
@RonEld @hanno-arm Thanks for your reviews! I've fixed the typo found by Hanno, but left the semi-colon in place as it was intentional - perhaps if we asked a technical writer to weigh in on the issue they would agree with Ron and explain me why it was not a good idea on the first place, but I think we can agree it's hardly worth it.

As this is a single-commit PR, I've taken the liberty to edit the commit rather than add one. Please review again.

**Comment by sbutcher-arm** [comment]
@RonEld - I've dismissed your review, as you seem to have come to an agreement on your points. If you'd like to review again and approve, please do so! Else we have enough approvers to promote the PR and merge it.

**Pull request merged**